src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -444,6 +444,40 @@ public static bool TryJoin(Span<char> destination, out int charsWritten, ReadOnl
             return true;
         }
 
+        public static bool TryJoin(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, ReadOnlySpan<char> path3, Span<char> destination, out int charsWritten)
```

Comment:
Any way this method can reuse the 2 path overload?

Join the first 2 paths, then combine the result with the 3rd?

Reply:
I do use the other overload. Part of the contract here is that I don't write into the buffer if I can't fit the results so I can't simplify beyond what I've done.